### ISSUE TITLE
Unexpected cursor position when using i_CTRL-o to add surroundings

### ISSUE THREAD

**Issue by kiryph** [machakann/vim-sandwich#67]
Right now vim-sandwich does not return to the desired cursor position in following example.

### Current behavior:
`word|` <kbd>Ctrl-o ysiw'</kbd> positions the cursor as following: `'|word'`

### Expected behavior:
`word|` <kbd>Ctrl-o ysiw'</kbd> should position the cursor as following: `'word'|`

`|` denotes the cursor position. I am using the vim-surround mappings described in the documentation.

I came across this issue on the vim-surround issue tracker:
https://github.com/tpope/vim-surround/issues/253

**Comment by machakann** [comment]
Good catch, thank you! I reproduced the problem. It seems different if not using `vim-surround` style mapping.

I will check it in this weekends

**Comment by machakann** [comment]
It seems difficult to fix this currently. The current behavior is correct in normal mode. Thus, we need to implement different logic for cursor placement.

The problem is there is no way to realize the use of <kbd>Ctrl-o</kbd> inside a script. Probably, we need this patch.
vim/vim#3000

**Comment by bew** [comment]
Hello! Just a friendly ping as the patch you mentionned in your last message has been merged
